backend/src/models/service.ts

```diff
@@ -5,11 +5,34 @@ import {
   PrimaryGeneratedColumn,
   ManyToOne,
   BaseEntity,
-  CreateDateColumn
+  CreateDateColumn,
+  BeforeInsert,
+  BeforeUpdate
 } from 'typeorm';
 import { Domain } from './domain';
 import { Scan } from './scan';
 
+interface Product {
+  // Common name
+  name: string;
+  // Product name
+  product?: string;
+  // Product vendor
+  vendor?: string;
+  // Product version
+  version: string;
+  // Product version revision
+  revision?: string;
+  // CPE without version (unique identifier)
```

Comment:
Why without version? I thought we would include version here if available?

Reply:
I think including without version will make it easier to uniquely identify products from different tools. We store version separately and use that when constructing the full CPE.